### PR TITLE
Improve ParallelBoundary code

### PR DESCRIPTION
**Pull request by dschwoerer**
Builds on #2629 

Allows for handling regions of short connection length differently from region of long connection length

This allows to use higher order stencil in general, without breaking for the points that don't support this.

### CODE REVIEW

**Comment by dschwoerer** [comment]
Requires #2640

**Comment by ZedThree** [comment]
This looks massive, but I guess it's because it includes `next` or something into the base branch? @dschwoerer Please could you update the base branch so the changes here are easier to see?

**Comment by dschwoerer** [comment]
My bad, I had forgotten this was a PR against par-bc-cleanup, not next. Thus merging next did only make things worse. Now I merged in par-bc-cleanup, and it is down to 16 commits.

File: include/bout/parallel_boundary_region.hxx [hunk 1]
```diff
@@ -46,28 +84,122 @@ class BoundaryRegionPar : public BoundaryRegionBase {
...
+      return neumann_o1(f, value);
+    }
+    ynext(f) =
+        parallel_stencil::neumann_o3(1 - length(), value, 1, f[ind()], 2, yprev(f));
+  }
 
   const int dir;
+
+private:
+  const BoutReal small_value = 1e-2;
```
[the remark this answers is not in the document]
Comment (ZedThree) [reply include/bout/parallel_boundary_region.hxx:191]:
```suggestion
  constexpr static BoutReal small_value = 1e-2;
```

**Review by bendudson — APPROVED** [review]

**Review by bendudson — APPROVED** [review]

**Pull request merged**